### ISSUE TITLE
There should be documentation on MutableFile upload events and callback

### ISSUE THREAD

**Issue by aeroxy** [qgustavor/mega#61]
```
    fetch({
      url,
      gzip: true
    }).on('error', e => {
      ...
    }).on('response', res => {
      ...
    }).pipe(
      megaStream
        .upload(filename)
        .on('complete', () => {
          ...
        })
        .on('error', e => {
          ...
        })
    );
```
From the current documentation there is only 'complete' event <https://github.com/qgustavor/mega/wiki/MutableFile> available (and I assume error is also available) which doesn't seem to be the case - some files are not uploaded successfully but they do not trigger either complete or error function...

**Comment by qgustavor** [comment]
The upload function [creates a stream here](https://github.com/qgustavor/mega/blob/170cbd7181a83ee3ab5d71308b580932a2378968/lib/mutable-file.js#L241) and [returns it here](https://github.com/qgustavor/mega/blob/170cbd7181a83ee3ab5d71308b580932a2378968/lib/mutable-file.js#L265) [Errors are handled here](https://github.com/qgustavor/mega/blob/170cbd7181a83ee3ab5d71308b580932a2378968/lib/mutable-file.js#L358-L365). The complete event [is handled here](https://github.com/qgustavor/mega/blob/170cbd7181a83ee3ab5d71308b580932a2378968/lib/mutable-file.js#L209).

[As show here](https://github.com/qgustavor/mega/blob/30f5588f4ccfae832954052b782088777859a817/test/storage.test.js#L88-L103) those events must work correctly. In the other hand there is no test case to check if the error event is working correctly. Is possible to implement such test case [using this feature](https://github.com/qgustavor/mega-mock/blob/fea813919f844fa540f954c5099f1a88ed1611b0/commands/handleUpload.js#L5-L8) of the API mocking library.

Considering that only today I tried uploading some files and it failed three times with `ERATELIMIT` errors. I think the error event is working correctly, but seems there is some change MEGA servers that's causing those errors. One thing to do is checking if there are other open-source MEGA clients that are also having issues and check what they did to fix those.

**Comment by qgustavor** [comment]
I updated the documentation clearing that the upload function returns a writeable stream. The upload issue isn't fixed yet but I will close this in favor of #60.
